include/boost/geometry/strategies/transform/matrix_transformers.hpp

```diff
@@ -41,6 +44,102 @@ namespace boost { namespace geometry
 namespace strategy { namespace transform
 {
 
+namespace detail { namespace matrix_transformer
+{
+
+template
+<
+    typename CalculationType,
+    typename P2,
+    std::size_t Dimension2,
+    std::size_t Dimension2Count
+>
+struct set_point_from_vec
+{
+    typedef CalculationType ct;
+    static inline void apply(P2& p2, qvm::vec<ct, Dimension2 + 1>& p2temp)
+    {
+        typedef typename geometry::coordinate_type<P2>::type ct2;
+        set<Dimension2Count>(p2, boost::numeric_cast<ct2>(qvm::A<Dimension2Count>(p2temp)));
+        set_point_from_vec<CalculationType, P2, Dimension2, Dimension2Count-1>::apply(p2, p2temp);
+    }
+};
+
+template
+<
+    typename CalculationType,
+    typename P2,
+    std::size_t Dimension2
+>
+struct set_point_from_vec<CalculationType, P2, Dimension2, -1>
+{
+    typedef CalculationType ct;
+    static inline void apply(P2& p2, qvm::vec<ct, Dimension2 + 1>& p2temp) {}
+};
+
+template
+<
+    typename CalculationType,
+    typename P1,
+    std::size_t Dimension1,
+    std::size_t Dimension1Count
+>
+struct set_vec_from_point
+{
+    typedef CalculationType ct;
+    static inline void apply(const P1& p1, qvm::vec<ct, Dimension1 + 1>& p1temp)
+    {
+        qvm::A<Dimension1Count>(p1temp) = get<Dimension1Count>(p1);
+        set_vec_from_point<CalculationType, P1, Dimension1, Dimension1Count - 1>::apply(p1, p1temp);
+    }
+};
+
+template
+<
+    typename CalculationType,
+    typename P1,
+    std::size_t Dimension1
+>
+struct set_vec_from_point<CalculationType, P1, Dimension1, -1>
+{
+    typedef CalculationType ct;
+    static inline void apply(const P1& p1, qvm::vec<ct, Dimension1 + 1>& p1temp) {}
+};
+
+
+template
+<
+    typename CalculationType,
+    std::size_t Dimension1,
+    std::size_t Dimension2
+>
+class matrix_transformer
+{
+protected :
+    typedef CalculationType ct;
+    typedef boost::qvm::mat<ct, Dimension2 + 1, Dimension1 + 1> matrix_type;
+    matrix_type m_matrix;
+public :
+
+    matrix_type const& matrix() const { return m_matrix; }
+    template <typename P1, typename P2>
+    inline bool apply(P1 const& p1, P2& p2) const
+    {
+        assert_dimension<P1,Dimension1>();
+        assert_dimension<P2,Dimension2>();
```

Comment:
A better place for assertions would probably be exactly where the coordinates are accessed, e.g. in `set_point_from_vec`.

Reply:
I did not spend much thought on the locations of these assertions when I wrote this, but after thinking about it now I think that if the user sees the asserts in that location failing, it would be more instructive than in the implementation details because as they are now, they communicate "you have the wrong transformer, it needs to match your points' dimension like in this assertion." (or this is what I hope they communicate).